pyfesom2/transport.py

```diff
@@ -0,0 +1,1083 @@
+                                  1000
+                                  )
+
+        # Bring the section coordinates into the disired shape [[],...,[]]
+        section_waypoints = [[section_lon[i], section_lat[i]] for i in range(len(section_lat))]
+
+    return section_waypoints, mesh, section
+
+def _ReduceMeshElementNumber(section_waypoints, mesh, section, add_extent):
+    '''
+    reduce_element_number.py
+
+    Reduces the number of elements that are loaded into memory to those nearby the section
+
+    Inputs
+    ------
+    section_waypoints (list)
+        list with all section waypoints [[lon,lat], ... ,[lon,lat]]
+    mesh (fesom.mesh object)
+        fesom.mesh
+    section (dict)
+        section dictionary
+    add_extent (int, float)
+        extent in degree that is added to the box cutout to contain all elements needed, choose small in case of high resolution meshes
+
+    Returns
+    -------
+    elem_box_nods (list)
+        list of indices that define the three nods of each element that belongs to the box
+    elem_box_indices (list)
+        list of indices where no_nan_triangles == True (to select the right elements when loading the data)
+    '''
+
+    # write section longitude and latitude in separate lists
+    section_waypoints_lon = [section_waypoints[i][0] for i in range(len(section_waypoints))]
+    section_waypoints_lat = [section_waypoints[i][1] for i in range(len(section_waypoints))]
+
+    if add_extent < 1:
+        warnings.warn(
+            'The extend added to the box is small, this might lead to errors when using low resolution meshes')
+
+    # find the maximum and minumum zonal/ meridional extent of the section
+    box_mesh = [min(section_waypoints_lon) - add_extent,
+                max(section_waypoints_lon) + add_extent,
+                min(section_waypoints_lat) - add_extent,
+                max(section_waypoints_lat) + add_extent
+                ]
+
+    # find the elements that are within the extent
+    elem_no_nan, no_nan_triangles = cut_region(mesh, box_mesh)
+    no_cyclic_elem2 = get_no_cyclic(mesh, elem_no_nan)
+    elem_box_nods = elem_no_nan[no_cyclic_elem2]
+
+    # create an array containing the indices of the elements that belong to the region
+    elem_box_indices = np.arange(mesh.e2d)[no_nan_triangles]
+
+    # Compute the distance of each section coodinate to the center of each element to further reduce the amount of polygons needed
+    # in case of meridional or zonal section the chosen box is already small enough to be loaded and no further elements have to be removed
+    # in all other cases the rectangular box gets to large and needs further shrinking
+    if section['orientation'] == 'other':
+        min_dist = add_extent * 100  # minimum distance in km to take element into account
+        distance_bool = list()
+
+        # compute the center of each element
+        element_center_lon = np.mean(mesh.x2[elem_box_nods], axis=1)
```

Comment:
In principle there is https://github.com/FESOM/pyfesom2/blob/b0782c3b1aab6a1d41f6f5a5bf166710b9c21cab/pyfesom2/ut.py#L641 You aproach will work most of the time as long as you don't use elements close to 180 degrees, so I would keep it for a moment :) We have to provide this data as part of the `fesom.mesh.diag.nc`